# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/common/system_utils.h
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/common/system_utils.h
--- a/gfx/angle/checkout/src/common/system_utils.h	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/common/system_utils.h	Sat Sep 07 13:38:36 2019 +0000
@@ -24,10 +24,13 @@
 std::string GetEnvironmentVar(const char *variableName);
 const char *GetPathSeparator();
 bool PrependPathToEnvironmentVar(const char *variableName, const char *path);
+bool IsDirectory(const char *filename);
 
 // Run an application and get the output.  Gets a nullptr-terminated set of args to execute the
 // application with, and returns the stdout and stderr outputs as well as the exit code.
 //
+// Pass nullptr for stdoutOut/stderrOut if you don't need to capture. exitCodeOut is required.
+//
 // Returns false if it fails to actually execute the application.
 bool RunApp(const std::vector<const char *> &args,
             std::string *stdoutOut,
@@ -48,7 +51,21 @@
     }
 };
 
-Library *OpenSharedLibrary(const char *libraryName);
+// Use SYSTEM_DIR to bypass loading ANGLE libraries with the same name as system DLLS
+// (e.g. opengl32.dll)
+enum class SearchType
+{
+    ApplicationDir,
+    SystemDir
+};
+
+Library *OpenSharedLibrary(const char *libraryName, SearchType searchType);
+
+// Returns true if the process is currently being debugged.
+bool IsDebuggerAttached();
+
+// Calls system APIs to break into the debugger.
+void BreakDebugger();
 }  // namespace angle
 
 #endif  // COMMON_SYSTEM_UTILS_H_